tests/integration/features/istio/production/upgrade-suite/istio_upgrade.feature

```diff
@@ -0,0 +1,19 @@
+Feature: Upgrade Istio
+
+  Scenario: Upgrade from current release to last version
```

Comment:
The reconciliation is never executed for the controller version and running this test will leave istio resources present on cluster. 
I think this is due to a timing issue with the current controller version being deployed.
The test waits until the `istio-controller-manager` is `ready`. After this it continues with verifying resource states and versions. The reconciliation of istio CR didn't start yet, because the controller is still starting and doing leader election, the status of Istio CR didn't update. So the test is not working as intended since the reconciliation never runs in the new controller, but the test successfully finishes.

Reply:
This is the case only if the Istio resources have correct version, otherwise the step with version checking will retry as many times as necessary for reconciliation to finish and pass if the resources got changed to required version or fail if the count of attempts will exceed the limit which somehow simulates situation when the upgrade failed.